### ISSUE TITLE
Add schema.org markup for datasets 

### ISSUE THREAD

**Issue by lwrubel** [sul-dlss/purl#796]
Add schema.org markup to support datasets being picked up by Google Search for its Dataset Search. This should only be added to purl pages where the item is a dataset. @arcadiafalcone is mapping Public XML to Google's set of recommended schema.org fields. 

An object is a dataset if has a `<genre>` tag with the value `dataset`.

More info:

- Analysis on schema.org for Datasets with links to Google's documentation: https://docs.google.com/document/d/1BO10k_zSTqqT1YmlCg5oE4tOsGXiqmHzQyb6itZypwo/edit
- [SearchWorks implementatio](https://github.com/sul-dlss/SearchWorks/blob/f2ed40371750c9d31bfba0b2f48f4dbd9d76de79/app/models/solr_document.rb#L36)n (note that it does not include all required fields). 
- [Example schema.org markup](https://developers.google.com/search/docs/appearance/structured-data/dataset#example)  from Google

**Comment by lwrubel** [comment]
See https://github.com/sul-dlss/purl/issues/794 for sample markup based on @arcadiafalcone's mapping in https://docs.google.com/spreadsheets/d/1vnDMTkzj4PeZql2W0m_MyTnjPQsTMnCKsmgJ-ne5QAc/edit#gid=0

**Comment by amyehodge** [comment]
I have compiled a list of datasets that I would expect to be found by Google Dataset Search at https://docs.google.com/spreadsheets/d/1zIZDlThxQIfy-LDQA8sx28De6q64Cj-94LfPEbErqt4/edit#gid=0. I've searched for all of these at Google Dataset Search via the titles and have indicated in column F whether I found them or not. Only 3 were found, but they came from other sources -- OSTI.gov, DataCite, or Zenodo.
